### PR TITLE
Return errors from enclave DB, instead of ignoring or using a critical log message

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

The enclave DB swallows a lot of errors, and uses a critical logger message for others. For resilience, we should return errors and handle those.

This is just the first step - there's too much to change to do it all in one PR.

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/enclave/db/storage.go [hunk 1]
```diff
@@ -79,11 +77,11 @@ func (s *storageImpl) StoreRollup(rollup *core.Rollup) {
 
 func (s *storageImpl) FetchRollup(hash common.L2RootHash) (*core.Rollup, bool) {
 	s.assertSecretAvailable()
-	r := obscurorawdb.ReadRollup(s.db, hash, s.logger)
-	if r != nil {
-		return r, true
+	rollup, err := obscurorawdb.ReadRollup(s.db, hash, s.logger)
+	if err != nil {
+		return nil, false
```
Comment (otherview) [inline go/enclave/db/storage.go:82]: Should this be logged ?

**Review by otherview — COMMENTED** [review]
Minor comments

File: go/enclave/db/rawdb/accessors_chain.go [hunk 2]
```diff
@@ -136,13 +139,17 @@ func ReadBodyRLP(db ethdb.KeyValueReader, hash gethcommon.Hash, number uint64, l
 	return data
 }
 
-func ReadRollupsForHeight(db ethdb.Database, number uint64, logger gethlog.Logger) []*core.Rollup {
+func ReadRollupsForHeight(db ethdb.Database, number uint64, logger gethlog.Logger) ([]*core.Rollup, error) {
 	hashes := ReadAllHashes(db, number)
 	rollups := make([]*core.Rollup, len(hashes))
 	for i, hash := range hashes {
-		rollups[i] = ReadRollup(db, hash, logger)
+		rollup, err := ReadRollup(db, hash, logger)
```
Comment (otherview) [inline go/enclave/db/rawdb/accessors_chain.go:146]: If the rollup is not found should it `rollups[i] = nil` ?

Comment (joeldudley) [reply go/enclave/db/rawdb/accessors_chain.go:146]: I think it should all succeed or all fail.

**Pull request merged**